### ISSUE TITLE
MSSQL missing single quote in dbgenerated datetime with default value

### ISSUE THREAD

**Issue by Tony-Truand** [prisma/prisma#21618]
Cli version : 5.4.2
MS SQL : 2019

In MSSQL 
If you do a pull on a MSSQL database containing a datetime (type datetime2) column with a default value (ex: '1900-01-01 00:00:00')
The result definition in the schema file is 
```prisma
theDate DateTime @default(dbgenerated("1900-01-01 00:00:00")
```
but it should be quoted like
```prisma
theDate DateTime @default(dbgenerated("'1900-01-01 00:00:00'")
```
Otherwise the push will fail on a error "Incorrect syntax at '00'"

See atatched GIF for "vidéo"

Step to reproduce

1️⃣  Create table in MSSQL

```sql
CREATE TABLE NewTable (
	id int IDENTITY(1,1) NOT NULL,
	theDate datetime2 DEFAULT '1900-01-01 00:00:00' NOT NULL,
	CONSTRAINT NewTable_PK PRIMARY KEY (id)
);
```
2️⃣  prisma Pull schema
```sh
npx prisma db pull
```

3️⃣ Delete table NewTable 
```sql
DROP TABLE NewTable;
```
4️⃣  prisma Push schema
```sh
npx prisma db push
```
 Result : Error

![image](https://github.com/prisma/prisma/assets/7480192/25a92011-d68f-452e-8e21-2b5bcf71d49a)

Full gif video : 
![Prisma_MSSQL_DateTime_Pull_Push_Error](https://github.com/prisma/prisma/assets/7480192/cba27314-bfba-48df-8ff3-b2dd4749803c)

**Comment by janpio** [comment]
I can confirm the introspection result, and the error you describe.



> but it should be quoted like
> 
> ```prisma
> theDate DateTime @default(dbgenerated("'1900-01-01 00:00:00'")
> ```
> 
> Otherwise the push will fail on a error "Incorrect syntax at '00'"

This will indeed make it work as expected, but of course there should be no reason for the default value to be represented via `dbgenerated(...)`.

It probably _should_ look like this:
```prisma
model NewTable {
  id      Int      @id(map: "NewTable_PK") @default(autoincrement())
  theDate DateTime @default("1900-01-01 00:00:00", map: "DF__NewTable__theDat__24927208")
}
```

Unfortunately this does not validate:
```
Error: Prisma schema validation - (validate wasm)
Error code: P1012
error: Error parsing attribute "@default": Parse error: "1900-01-01 00:00:00" is not a valid rfc3339 datetime string. (input contains invalid characters)
  -->  prisma\schema.prisma:12
   | 
11 |   id      Int      @id(map: "NewTable_PK") @default(autoincrement())
12 |   theDate DateTime @default("1900-01-01 00:00:00", map: "DF__NewTable__theDat__24927208")
   | 
```

Can be avoided by adapting the default string:
```
model NewTable {
  id      Int      @id(map: "NewTable_PK") @default(autoincrement())
  theDate DateTime @default("1900-01-01T00:00:00Z", map: "DF__NewTable__theDat__24927208")
}
```

But unfortunately leads to slightly different migration and default value in the end:
```sql
-- CreateTable
CREATE TABLE [dbo].[NewTable] (
    [id] INT NOT NULL IDENTITY(1,1),
    [theDate] DATETIME2 NOT NULL CONSTRAINT [DF__NewTable__theDat__24927208] DEFAULT '1900-01-01 00:00:00 +00:00',
    CONSTRAINT [NewTable_PK] PRIMARY KEY CLUSTERED ([id])
);
```

---

TLDR: Using `@default(dbgenerated("'1900-01-01 00:00:00'")` right now is indeed the best way to achieve the desired result. Prisma needs to improve the handling of `Datetime` fields for SQL Server.

**Comment by hamzamihaidanielx** [comment]
This has been opened for quite a while, and its blocker nowadays too for those who use SQLServer. Any ideea when this will get tackled? Thank you!
